empire.go

```diff
@@ -141,6 +145,10 @@ func (opts CreateOpts) Event() CreateEvent {
 
 // Create creates a new app.
 func (e *Empire) Create(ctx context.Context, opts CreateOpts) (*App, error) {
+	if err := e.MaintenanceMode; err != nil {
```

Comment:
Another possibility is to do this as a decorator/middleware around the empire.Empire interface:

```go
type MaintenanceModeEmpire struct {
        *MaintenanceMode
        empire *Empire
}

func (e *MaintenanceModeEmpire) Create(ctx context.Context, opts CreateOpts) (*App, error) {
        if err := e.MaintenanceMode; err != nil {
                return nil, err
        }

        return e.empire.Create(ctx, opts)
}
```

So we would wrap the empire.Empire struct before passing to the server to consume. This would be a little more explicit, because you have to explicitly whitelist each method but ends up being more boilerplate, and it means that control layer packages need to accept an interface instead of a struct, which adds some indirection (but not that much). I can be sold either way.